### DISCUSSION TITLE
does strongswan support Linux VRF ?

### DISCUSSION THREAD

**Discussion by sprafull** [strongswan/strongswan#2142] [General]
does strongswan support Linux VRF ?

**Comment by tobiasbrunner** [comment]
In what way?

**Reply by sujalsheth** [reply]
I wish to configure IPSec tunnels with endpoints on different VRF.
AFAIK, strongswan recommends to create instance in different namespace to support this.
 
But now Linux kernel supports VRF (https://docs.kernel.org/networking/vrf.html), if strongswan integrate this changes then it can support above requirement with single instance.

**Reply by tobiasbrunner** [reply]
Maybe have a look at [this section](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html#_xfrm_interfaces_in_vrfs) in the docs.

**Reply by sujalsheth** [reply]
Thanks for reply. !!! I have read above link. 
But how does IKE Packets would be sent over specific IP Address on specfic vrf. where to configure that ?? i couldn't find any option in swanctl.conf

**Reply by tobiasbrunner** [reply]
You can specify a specific local address for each connection. But there are no options for VRFs in regards to IKE traffic.

**Reply by sujalsheth** [reply]
Is there roadmap to support IKE traffic over VRF ?

**Reply by tobiasbrunner** [reply]
Not at this time.

**Comment by csszep** [comment]
Hi!

Based on my tests with 6.0.1 , sending IKE packets from the correct VRF is already working; you just need to set the set_sourceif=yes parameter in the socket-default plugin. Additionally, the following sysctl variable must be configured: net.ipv4.udp_l3mdev_accept=1.

When using XFRM interface, ESP packets will also be sent from the correct VRF if you set the XFRM "dev PHYS_DEV" to the appropriate interface, which may differ from the VRF of the XFRM interface.

It would be helpful if this behavior were documented more explicitly in the strongSwan documentation.
